Honor '--without-cunit', addressed upstreams with https://github.com/pgpointcloud/pointcloud/pull/349

Only link to libxml2, addresses https://trac.macports.org/ticket/68704

--- configure.ac.orig
+++ configure.ac
@@ -138,7 +138,7 @@
   CPPFLAGS="${CPPFLAGS_SAVE}"
 fi
 
-if test "$FOUND_CUNIT" = "YES"; then
+if test "$FOUND_CUNIT" = "YES" && test x${with_cunit} != "xno"; then
   AC_DEFINE([HAVE_CUNIT], [1], [Have CUnit])
   CUNIT_STATUS="enabled"
   if test $CUNITDIR; then
@@ -281,7 +281,7 @@
 
 
 dnl Extract the linker and include flags
-XML2_LDFLAGS=`$XML2CONFIG --libs`
+XML2_LDFLAGS=`$XML2CONFIG --libs --dynamic`
 XML2_CPPFLAGS=`$XML2CONFIG --cflags`
 
 dnl Extract the version
